26057 vergelijkingstool

jcacosta89 opened the pull request:
**Description**

This PR introduces a significant update to the *vergelijkingstool* module, restructuring the codebase, refactoring the GUI, improving the comparison workflow, and adding support for summary tables and automated testing.
The main goal is to make the code more maintainable, more organized, and easier to extend, while also enabling reliable unit testing.

Below is a description of the updated modules, their responsibilities, and the improvements included in this PR.

 **1. config.py**

`config.py` now contains a set of independent lists used to select specific layers for each type of comparison (DAMO, HDB, 3Di).
This allows users (and the GUI) to choose only the relevant layers without hard-coding selections inside the comparison logic.

**2. DAMO.py**

The `DAMO` class reads all DAMO/HDB layers and stores them in a dictionary called `data`.
This file contains the core logic for comparing the old and updated DAMO datasets.
In this PR, the internal structure was cleaned up and made more consistent, especially around attribute comparison and derived columns.

**3. Dataset.py**

`Dataset` remains the shared base class for both DAMO and 3Di.
It handles reading, storing, and translating layers, and centralizes functionality that used to be duplicated across modules.

**4. main.py**

The main comparison workflow still lives here, but the logic is now much cleaner thanks to refactoring.
Several responsibilities—such as styling, exporting, and summary construction—were moved to dedicated modules, making the central comparison flow clearer and easier to test.

**5. styling.py**

All export and styling functions were extracted into this module.
This removes duplication from DAMO and 3Di classes and ensures that all QGIS styling rules are defined in one place.

**6. Threedimodel.py**

`Threedimodel` now provides a clearer interface for reading the 3Di schematisation SQLite and performing comparisons with DAMO.
Here we also find the core comparison logic for 3Di → DAMO.
The reading and translation logic was simplified and made more robust.

**7. utils.py**

This file received several key improvements:

* The new `ModelInfo` class centralizes all model paths (input and output).
  This avoids repetition and reduces path-related errors.

* Duplicate helper functions previously scattered between DAMO and 3Di were consolidated here.

* `load_file_and_translate` was introduced to replace the old SQLite reader.
  It now loads only GeoPackage files, consistent with modern model formats.

* `update_channel_code` was added to synchronize channel codes using DAMO hydroobject data together with 3Di channels and cross-sections.
  This ensures that comparisons based on `code` behave correctly.

* Two central summary-related functions were added:

  * **add_priority_summaries** builds row-level `Summary_Critical` and `Summary_Warnings`.
  * **build_summary_layers** generates a final summary table with counts of warnings/critical issues per element.

These functions make the comparison output much easier to interpret directly from QGIS.

**8. test_vergelijkingstool.py**

A full test suite was added.
It validates the core comparison functions, priority logic, summary generation, and the overall workflow using a real test dataset.
This prevents regressions and ensures future changes remain stable.

**9. json_files**

All JSON files were reorganized into this dedicated folder, improving clarity and avoiding stray files.
The most relevant one is `damo_attribute_comparison.json`, which defines all comparison rules.

**10. qml_styling_files**

All QGIS styles are now grouped here, one QML per layer/class.
This keeps the project organized and ensures consistent styling in QGIS outputs.

**11. gui/**

The GUI was completely refactored.
It now uses a dedicated class, `VergelijkingstoolGUI`, which makes the interface far easier to manage, extend and test.
Improvements include:

* Dynamic inputs (model paths, output paths, filenames).
* Clear distinction between comparing DAMO vs DAMO, DAMO vs 3Di, or Both.
* Layer and structure selection widgets that appear only when relevant.
* Clean handling of click callbacks, preventing duplicate behaviour.
* A structured, readable layout.

jcacosta89 committed:
first update from the vergelijking_tool.

This version contain a example from Castricum. Some hardcode have been made to make it work.

jcacosta89 committed:
Create 06_vergelijkingstool.ipynb

jcacosta89 committed:
update input_file

jcacosta89 committed:
Merge branch 'main' into Vergelijkingstool

jcacosta89 committed:
Update __init__.py

import all scripts used in the vergelijkingstool

jcacosta89 committed:
Update config.py

update layers names to be compared.

jcacosta89 committed:
Update DAMO.py

update DAMO Class, to be able to select or not the layers to be compared

jcacosta89 committed:
Update Dataset.py

jcacosta89 committed:
Update main.py

update main to select which function to used according to the selection done: It could be: compare DAMO WITH DAMO, and then select specific layer or select them all. It could be also  compare DAMO WITH 3DI, and then select specific layer or select them all. Or it is possible also to select with both.

jcacosta89 committed:
Update styling.py

update Styling to add: model name, and date.
update Styling to make transparent the on that fall in  both.

jcacosta89 committed:
Update Threedimodel.py

update styling and create specific layer if layer selection is TRUE

jcacosta89 committed:
Update 06_vergelijkingstool.ipynb

Set the new paths

jcacosta89 committed:
add to gitnore

jcacosta89 committed:
update name_date, add comments

jcacosta89 committed:
UPDATE DATASET TO MAKE LAYERS  SELECTABLE

A

jcacosta89 committed:
Update 06_vergelijkingstool.ipynb

jcacosta89 committed:
Merge branch 'main' into Vergelijkingstool

jcacosta89 committed:
updata vergelijking tool DAMO DAMO

jcacosta89 committed:
Update DAMO.py

jcacosta89 committed:
Update main.py

jcacosta89 committed:
Update name_date.py

jcacosta89 committed:
Update styling.py

jcacosta89 committed:
Update config.py

jcacosta89 committed:
Update config.py

jcacosta89 committed:
Update Threedimodel.py

jcacosta89 committed:
Update main.py

jcacosta89 committed:
unhash code needed

jcacosta89 committed:
add function export_compariso_new and delete export_comparions. Also re structure function to be call from styling.py

jcacosta89 committed:
deletestyling for points, lines and polygon.

 Working now with qml files.
Add function  geom_kind, to check what sort of geometry are we using,
add function replace label DAMO and replace label 3di, to update labels in symbology.
add function def prepare_layers_for_export(table_C, filename, overwrite=False) a share code shared by export comparision in DAMO and Threedi.

jcacosta89 committed:
Update Threedimodel.py

delete fucntion from sqlite, add fucntion from geopackge.
delete function def export_comparison() and add def export_comparison_3di() that calls   styling.export_comparison_3di(
            table_C, statistics, filename, overwrite=overwrite, styling_path=styling_path, crs=self.crs
        )

jcacosta89 committed:
update inputs and unary_union for union_all()

jcacosta89 committed:
Apply pre-commit fixes ruff

jcacosta89 committed:
Delete name_date.py

change it for restructure

jcacosta89 committed:
change import for restructure code

jcacosta89 committed:
update Dataset with ModelInfo initialzed

jcacosta89 committed:
Create Utils.py with class ModelInfo. Easier to get all the models information

jcacosta89 committed:
update inputs base on new class modelinfo

jcacosta89 committed:
update styling with new class model_info

jcacosta89 committed:
update import still need some work. Not good initialized yet

jcacosta89 committed:
Add parameters to call class model_info

jcacosta89 committed:
add docstring and delete hased lines

jcacosta89 committed:
Merge branch 'main' into Vergelijkingstool

jcacosta89 committed:
add input folder and json folder to the dataclass modelinfo

jcacosta89 committed:
remove not needed imports

jcacosta89 committed:
update inputs with model info.

jcacosta89 committed:
delete not needed function since it is not working with geopackge anymore.

jcacosta89 committed:
Remove function that is share with Threedimodel.py

jcacosta89 committed:
Remove not needed imports

jcacosta89 committed:
Remove not necesary imports. Remove function also use in DAMO. Initialized self.data with function coming from utils

jcacosta89 committed:
Add function that retrieve data from DAMO, HDB and 3di.

jcacosta89 committed:
update main.

jcacosta89 committed:
remove old styling, and add new splited in geometry and type of comparision DAMO and Threedi

jcacosta89 committed:
update symbology. Add symbology in case tehre is no column number_of_critical

jcacosta89 committed:
add utils to damo and threedi and update folder structure

jcacosta89 committed:
Update folders.py

jcacosta89 committed:
delete not necessary inputs

jcacosta89 committed:
Initialize  function in the class that update channel layer before the comparison process starts.  Add new info from model in the initialization to be used in the update_channel_codes

jcacosta89 committed:
add function update channel codes.

jcacosta89 committed:
ruff

jcacosta89 committed:
Update folders.py

jcacosta89 committed:
Update folders.py

jcacosta89 committed:
delete not necesary inputs

jcacosta89 committed:
Update main.py

jcacosta89 committed:
Update main.py

jcacosta89 committed:
remove shapes file with geometry issues

jcacosta89 committed:
remove layer without geometries.  fix docstrings

jcacosta89 committed:
update docstrings

jcacosta89 committed:
update stlyling and add polygons style

jcacosta89 committed:
update docstrings

jcacosta89 committed:
ruff

jcacosta89 committed:
Refactor styling file structure and usage

Moved QML styling files for DAMO and Threedi to new 'qml_styling_files' package and updated code to reference these new locations. Removed redundant 'styling_path' argument from several methods and main entrypoint, now using internal styling path resolution. Fixed label formatting in styling.py to use correct model info attributes.

jcacosta89 committed:
ruff

jcacosta89 committed:
read json files from repository

jcacosta89 committed:
ruff

jcacosta89 committed:
fix damo attribute comparison to be read from repository

jcacosta89 committed:
ruff

jcacosta89 committed:
update columns names in json file, update json comparison file 3di to be read from repository, make function max_value_from tabulated, update the values in the cross_section per layer where the column cross section table exists dele unnecesary function.  such: max_value_from_tabullated and join_cross_section_definition

jcacosta89 committed:
ruff

jcacosta89 committed:
Merge branch 'main' into Vergelijkingstool

jcacosta89 committed:
create init file

jcacosta89 committed:
move gui from ipynb to vergelijkngstool_gui.

Create fucntions so make the gui work in one flow
 1.  def on compare change
2. damo_checkbox_change
3. 3di_checkbox_change
4. on_run_clicked

jcacosta89 committed:
Update main.py

jcacosta89 committed:
ruff   :-(

jcacosta89 committed:
fix gui. update it with a new version.

jcacosta89 committed:
ruff

jcacosta89 committed:
version 2 fucntional gui

jcacosta89 committed:
Update Damo jsonfile and Damo..py

In here we are updating the columns, from a to new and from b to OLD so it will be related to the layer name.

jcacosta89 committed:
Update DAMO.py

jcacosta89 committed:
fix dataset, imporve condition for BOTH, add rules por duiker of peilgrens,  update dataset columns

jcacosta89 committed:
update inputs

jcacosta89 committed:
update attribute qml for bridges

jcacosta89 committed:
Fix docstrings

jcacosta89 committed:
Add comparision rules for duikers, hydro deelgebidede, hydroobject gemaal, gemaal op peilgrens.

jcacosta89 committed:
update symbologies per layers. Include update in Attribute tabels

jcacosta89 committed:
add comparison rules for hydroobject,  peilafwijkinggebied, peilgebiedpraktijk, sturing_3di, stuw, stuwen op peilegrens

jcacosta89 committed:
update symbology files

stuart-minion committed:
new branch for vergelijkingstool that includes v0 in history

jcacosta89 committed:
update damo atttibute comparasion json file: add rules for stuw, stuwen op peilgrens, vastedam and vispassages

jcacosta89 committed:
update symbology per feature.

jcacosta89 committed:
Merge branch 'Vergelijkingstool' into 26057-Vergelijkingstool

jcacosta89 committed:
Update config.py

jcacosta89 committed:
update json files and qml files

jcacosta89 committed:
update and fix json files and qml

jcacosta89 committed:
Update config.py

jcacosta89 committed:
Update main.py

jcacosta89 committed:
Update dataset file. make the function get_structure_code more clear

jcacosta89 committed:
Update model_attribute_comparison.json

jcacosta89 committed:
make neuw rules, and update symbology including: attribute tables, symbology and labels.

jcacosta89 committed:
Update main.py

jcacosta89 committed:
updte how it read jsonfiles

jcacosta89 committed:
change the input layer on the code to be read from input_oud_data

jcacosta89 committed:
update gui options. Make it more clear

jcacosta89 committed:
update layer selection, make it smaller and easier.

jcacosta89 committed:
Update main with comparision names, update threedimodel: Add threedi layer and damo layer in the initialization.

jcacosta89 committed:
update json file. It will avoid to bee seen in the attribute table incosistenci in the category

jcacosta89 committed:
1. Add fucntion that creates fields: Summary Critical and Sumaary Warnings. in Utils. Update damo and threedimodel.py to call that function. Update json file to delete priority column information., Upadate main with new inputs options.

jcacosta89 committed:
DAMO: Add Report box to the attribute table style

jcacosta89 committed:
Update main.py

jcacosta89 committed:
Update main.py

jcacosta89 committed:
Update Rules for Threedi - DAMO, add summary columns and improve the attribute table form/layout

jcacosta89 committed:
change condition in the rule base symbology. De colums ws_category did not exist. change for categorieoppwaterlichaam

jcacosta89 committed:
update simbology damo, make it dinamic. Make color red to columns that have critcal values and orange for warning

jcacosta89 committed:
make simbology for 3di dinamic. red tabs for critical orange tab for warnings.

jcacosta89 committed:
summary per layer. correct damo comparison, change to category.

jcacosta89 committed:
update dateaset and utils

jcacosta89 committed:
update summary table, add vergelijking rules for 3di_model and update vergelijkings rule for DAMO,

jcacosta89 committed:
Merge branch 'main' into vergelijkingstool_pytests

jcacosta89 committed:
Update config.py

jcacosta89 committed:
add pyhints

jcacosta89 committed:
add docstrings, pyhints  and comments. Update gitignoe

jcacosta89 committed:
Update damo_attribute_comparison.json

jcacosta89 committed:
Add waterdeel qml for stylings

jcacosta89 committed:
remove not necessary function that were used before in first styling versions.

jcacosta89 committed:
Create test_vergelijkingstool.py

jcacosta89 committed:
delete files. not needd

jcacosta89 committed:
update main

jcacosta89 committed:
update gitignore

jcacosta89 committed:
Merge branch 'vergelijkingstool_pytests' into 26057-Vergelijkingstool

jcacosta89 committed:
delete temporal path use tmp_path fixture from pytest. update utils, go case where there is no critcial, add temporarily column to waterdeel tables. later it is not use.

jcacosta89 committed:
update gui

jcacosta89 committed:
Update .gitignore

jcacosta89 committed:
Update Threedimodel.py

jcacosta89 committed:
cambiar gdb por gpkg

jcacosta89 committed:
Create polder_polygon.gpkg

jcacosta89 committed:
add database to compare

jcacosta89 committed:
improve test

jcacosta89 committed:
add data to the model including output data

jcacosta89 committed:
add bwn_test model again. and resutls models to test

jcacosta89 committed:
update files

jcacosta89 committed:
Update structure_control.py

jcacosta89 committed:
Merge branch 'main' of https://github.com/threedi/hhnk-threedi-tools

stuart-minion commented on hhnk_threedi_tools/core/vergelijkingstool/Threedimodel.py:
```diff
@@ -1,215 +1,119 @@
 
 class Threedimodel(DataSet):
-    def __init__(self, filename, translation=None):
+    def __init__(self, filename: Union[str, Path], model_info: ModelInfo, translation: Optional[dict] = None) -> None:
```
In de init van class Treedimodel is een parameter translation, maar die wordt niet gebruikt. Waarom?

jcacosta89 replied:
Initialization class has changed. Now it get the translation file path from the class ModelInfo

stuart-minion reviewed (commented):
Hi Juan,

ik kan de tool draaien en krijg de juiste output. Het is heel veel dus ik kan niet van alles beoordelen of het goed gaat. Ik vind de output heel prettig werken met de opmaak en de opdeling van de tabellen, je krijgt daarmee een goed overzicht.
<img width="1809" height="1075" alt="image" src="https://github.com/user-attachments/assets/9b5e3db3-52eb-46e4-811e-c498cf6ef984" />

Van het lezen van de gegevens vraag ik me af of dit nu niet dubbel is met wat we al hebben in threedi of research tools, dus de classes Threedimodel en DAMO. Zou het mogelijk zijn deze te integreren. @wvangerwen wat denk jij?

Je hebt wel heel veel gedaan aan docstring en typehints, dat is mooi. 

En Juan zou jij een afspraak in willen plannen om met mij en Wietse nog een keer naar de hele code te kijken? Ik weet bijvoorbeeld niet waar of hoe ik de gui kan starten. Wellicht is een stukje readme en documentatie daarbij nog nodig.

Groet,
Wouter

jcacosta89 committed:
Merge branch 'main' of https://github.com/threedi/hhnk-threedi-tools

jcacosta89 committed:
update model path

jcacosta89 committed:
fix output file

jcacosta89 committed:
Revert "Update structure_control.py"

This reverts commit 7b839fa2eaeca1bc9706bb30d5f00f69cbca307b.

jcacosta89 committed:
update columns in models, damo and hdb. Update table's names,  Update translation files,

jcacosta89 committed:
update function update_channel_codes. Take the codes from damo an place it in the model.

This is a bite tricky. Normally  hydroobject in damo are longer than channels in models. Hydroobject are normally divided by connection nodes to later on added to the model. That is why we are going to find feature with the same code in table OAF.

jcacosta89 committed:
Delete 06_vergelijkingstool_v2.ipynb

jcacosta89 committed:
update vergelijkingstool DAMO

Add following updates:
- Change de name in HDB from duikersifonhevel to Duikers
- Remove empty table from dictionary Table_C
- Skip column not found in the table from comparision
- Update damo_translation file: "bodemhoogteNAP": "bodemhoogte_nap"

jcacosta89 committed:
Add number of critical and warning as summary

jcacosta89 committed:
Add number of critical summary

jcacosta89 committed:
Update styling. If there are 2 geometry type in the table_C (model: orifice-> line, damo:bridge -> point) keep the point.  Update styling  to adapt it to points.

jcacosta89 committed:
Add waterway category to the layers that come from 3di and thay are not in DAMO

jcacosta89 committed:
make a buffer to the polder polygon to be sure that it collect all the data in the model. Update simbology in KBR and label in KDU

jcacosta89 committed:
Merge branch 'main' into 26057-Vergelijkingstool

jcacosta89 committed:
update vergelijkings tool notebook

jcacosta89 committed:
update hdb list layers. add polderpolygon in main, add function for hdb name selection.

jcacosta89 committed:
update function. change name

jcacosta89 committed:
Add readme file to verschelijkingtool

stuart-minion committed:
opmerkingen readme

jcacosta89 committed:
Update new vesion a bit more clear and specific.

jcacosta89 committed:
Update README.md

jcacosta89 committed:
update Folder Structure, update symbology por KDU points, update readme, update styling function, update, model info class with new names.

jcacosta89 committed:
Update styling.py

jcacosta89 committed:
add word file to the docs and add a fucntion that copy the file in the folder.  change folder structure

jcacosta89 commented on hhnk_threedi_tools/core/vergelijkingstool/config.py:
```diff
@@ -1,28 +1,32 @@
 
-SQLITE_LAYER_CROSS_SECTION_DEFINITION = "v2_cross_section_definition"
+# Name of the table containing cross-section definitions
+SQLITE_LAYER_CROSS_SECTION_DEFINITION: str = "v2_cross_section_definition"
```
geupdated

wvangerwen commented on tests/test_vergelijkingstool.py:
```diff
@@ -0,0 +1,435 @@
+# %%
+import json
+import os
```
removed.

wvangerwen commented on tests/test_vergelijkingstool.py:
```diff
@@ -0,0 +1,435 @@
+from hhnk_threedi_tools.core.vergelijkingstool.utils import ModelInfo, get_model_info
+from tests.config import FOLDER_TEST
+
+# model_info = get_model_info(FOLDER_TEST)
```
removed

jcacosta89 commented on hhnk_threedi_tools/core/vergelijkingstool/main.py:
```diff
@@ -17,7 +17,11 @@
 Unpack content of .7z file and place them in the C:\\Windows\\System32 folder
```
done

jcacosta89 commented on hhnk_threedi_tools/core/vergelijkingstool/main.py:
```diff
@@ -17,7 +17,11 @@
+# %%
+__authors__ = [
+    "Thijs van den Pol (Royal HaskoningDHV)",
+    "Emiel Verstegen (Royal HaskoningDHV)",
```
Notice.md file create at vergelijkingstool/docs/NOTICE.md
# Notice – Vergelijkingstool

Deze vergelijkingstool is mede ontwikkeld door Royal HaskoningDHV in opdracht van Hoogheemraadschap Hollands Noorderkwartier (HHNK).
Het gebruik van deze tool is bedoeld voor HHNK. Voor gebruik, verspreiding of aanpassing door andere partijen dan HHNK dient vooraf toestemming te worden gevraagd aan HHNK.


authors:Thijs van den Pol (Royal HaskoningDHV), Emiel Verstegen (Royal HaskoningDHV),

contact = "emiel.verstegen@rhdhv.com"
credits = ["Thijs van den Pol", "Emiel Verstegen"]
date = "2023/03/13"
deprecated = False
email = "emiel.verstegen@rhdhv.com"
maintainer = "developer"
status = "Production"
version = "1.1.0"

jcacosta89 commented on hhnk_threedi_tools/__init__.py:
```diff
@@ -39,6 +39,9 @@
     update_bank_levels_last_calc,
 )
 
+# backup
```
remove

jcacosta89 commented on tests/test_vergelijkingstool.py:
```diff
@@ -0,0 +1,435 @@
+
+
+@pytest.mark.skipif(sys.version_info < (3, 12), reason="Requires Python 3.12 or higher")
+def test_pprepare_layers_for_export(damos: Tuple[DAMO, DAMO], model_info: ModelInfo) -> None:
```
function name changed  to def test_specific_layer_for_export

jcacosta89 commented on tests/data/model_test/01_source_data/polder_polygon.gpkg:
fix bug in main:
```
 if not fn_DAMO_selection.exists():
        # change to gpkg
        fn_shp = fn_DAMO_selection.with_suffix(".shp")
        gdf = gpd.read_file(fn_shp, engine="pyogrio")
        gdf.to_file(fn_DAMO_selection, driver="GPKG")
```

jcacosta89 commented on hhnk_threedi_tools/core/vergelijkingstool/config.py:
```diff
@@ -33,43 +37,53 @@
     "VasteDam",
     "Vispassage",
     "Waterdeel",
+    # "Waterdeel", it does not contains code should be done in a different way
```
removed

jcacosta89 commented on hhnk_threedi_tools/core/vergelijkingstool/dataset/threedimodel.py:
ja

jcacosta89 committed:
update test data. folder structure in test data. Improve code to remove warnings.  Change location damo, threedi and datasete to dataset folder within vergelijkingstool. Add Notice.md to add credits.

jcacosta89 committed:
ruff

jcacosta89 committed:
Add some comment to the tool to understand better. Change result for the pytest.

jcacosta89 committed:
update hints for pytests

jcacosta89 committed:
update file location and json files

remove: Doorstroomopening because it is already in the brug database
update: dataset and damo location. Re located to have a better reading.
update: config.py to ad cross_section, aquaductilijn and vispassge
update json file to cahnge proident from cross section to code

jcacosta89 committed:
ruff

jcacosta89 committed:
update log. Error was too dramatic. It is just a warning

jcacosta89 committed:
Update damo_attribute_comparison.json

jcacosta89 commented on tests/test_vergelijkingstool.py:
```diff
@@ -0,0 +1,435 @@
+    filename = model_info.output_folder / "DAMO_comparison.gpkg"
+
+    # run compare and write gpkg
+    damo_new.compare_with_damo(damo_old, filename=filename, overwrite=True)
```
logs error and bugs removed as they were not needed

jcacosta89 committed:
Merge branch 'main' into 26057-Vergelijkingstool

jcacosta89 committed:
MERGE WITH MAIN

The merge does not include anything from the vergelijkstool. The enviroment was failing too much. It did not work at all. That is why I decided to merge MAIN into vergelijkingstool.

jcacosta89 committed:
update comments fro PR

1. Update layer name in damo config file
2. Delete comparition nae attrbiutes for hydroobject
3. Add new comparision rules for sluis
4.  update json translation file for hydroobjectes
5. Add qml file for sluis, and update sturing kunstwerken qml file
6. Update logger message for the styling.

stuart-minion commented:
> Hi Juan,
> 
> ik kan de tool draaien en krijg de juiste output. Het is heel veel dus ik kan niet van alles beoordelen of het goed gaat. Ik vind de output heel prettig werken met de opmaak en de opdeling van de tabellen, je krijgt daarmee een goed overzicht. <img alt="image" width="1809" height="1075" src="https://private-user-images.githubusercontent.com/48908991/536121641-9b5e3db3-52eb-46e4-811e-c498cf6ef984.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3ODU0ODg2MTMsIm5iZiI6MTc4NTQ4ODMxMywicGF0aCI6Ii80ODkwODk5MS81MzYxMjE2NDEtOWI1ZTNkYjMtNTJlYi00NmU0LTgxMWUtYzQ5OGNmNmVmOTg0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNjA3MzElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjYwNzMxVDA4NTgzM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQxODJhNTRhMTAyOTM0YTI3NTc3MmJkNTZkMjBkZjM5ZWExNDhkNjI4NThlMGEyMWM1NDBlNjMyODc4ZDhmYzMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JnJlc3BvbnNlLWNvbnRlbnQtdHlwZT1pbWFnZSUyRnBuZyJ9.hboXdkoTxs_zavslgB777pZNzV_oQfOw9kBCp8zRuQc">
> 
> Van het lezen van de gegevens vraag ik me af of dit nu niet dubbel is met wat we al hebben in threedi of research tools, dus de classes Threedimodel en DAMO. Zou het mogelijk zijn deze te integreren. @wvangerwen wat denk jij?
> 
> Je hebt wel heel veel gedaan aan docstring en typehints, dat is mooi.
> 
> En Juan zou jij een afspraak in willen plannen om met mij en Wietse nog een keer naar de hele code te kijken? Ik weet bijvoorbeeld niet waar of hoe ik de gui kan starten. Wellicht is een stukje readme en documentatie daarbij nog nodig.
> 
> Groet, Wouter

In the new corebusiness structure we will refactor the DAMO and ThreediModel classes/structure.

stuart-minion commented on hhnk_threedi_tools/core/vergelijkingstool/config.py:
```diff
@@ -1,28 +1,32 @@
+    # "Bergingsgebied",  # Niet in beide datasets
     "Brug",
-    "Doorstroomopening",
+    # "Doorstroomopening", not a gdf
```
https://dev.azure.com/HHNK/Intern/_workitems/edit/28861

stuart-minion commented on hhnk_threedi_tools/core/vergelijkingstool/config.py:
```diff
@@ -1,28 +1,32 @@
-    "GW_PRO",
-    "GW_PRW",
+    # "GW_PBP", no gdf
+    # "GW_PRO",
```
done

stuart-minion commented on hhnk_threedi_tools/core/vergelijkingstool/config.py:
```diff
@@ -1,28 +1,32 @@
     "AfvoergebiedAanvoergebied",
-    "AquaductLijn",  # Niet in beide datasets
-    "Bergingsgebied",  # Niet in beide datasets
+    # "AquaductLijn",  # Niet in beide datasets
```
done

jcacosta89 committed:
update gw_pro

1. create json comparison for this layer
2. Add a fucntion that create de watercategory column, to make the simbology work
3. Create symbology style

jcacosta89 committed:
add aquaductijn set

add: qml file for styling,  jsonfile, config

jcacosta89 committed:
Update layers

Add qml files for: Polderclusters and bergingsgebied
Add json comparison for polderclusters.
Add Bergingsgebied and IWS_GEO_BESCHR_PROFIELPUNTEN in the config file to be compared.
Change columns names for poldercluster and iws geo beschr profielpuntne

jcacosta89 committed:
Add profile punts layers

Add profile points layers.
Add function to add category to the layer
Add qml file
Add comparison to the json file
Add translation file to change column name

jcacosta89 committed:
update qml file.

jcacosta89 committed:
ruff

jcacosta89 committed:
update missing qml file for DAMO cmp

jcacosta89 committed:
Merge branch 'main' into 26057-Vergelijkingstool

jcacosta89 committed:
Update symbology and gemaal units

jcacosta89 committed:
chnage gui_helpers per gui_helpers.py

chnage gui_helpers per gui_helpers.py

jcacosta89 committed:
Update gui_helpers.py

jcacosta89 committed:
Update vergelijkingstool_gui.py

jcacosta89 committed:
Create vergelijkingstool_guide.md

jcacosta89 committed:
update readme and gui

jcacosta89 committed:
Update vergelijkingstool_guide.md

jcacosta89 committed:
Update vergelijkingstool_guide.md

jcacosta89 commented:
First PR finished.
